### DISCUSSION TITLE
Inconsistency between pyright in githubactions vs locally

### DISCUSSION THREAD

**Discussion by mjperrone** [microsoft/pyright#5916] [Q&A]
## Context

I started this journey just trying to get the `pyright` cli command to output the same errors as the vscode pylance extension. I found some threads docs and repos to help me with that:

* https://github.com/microsoft/pylance-release/blob/main/USING_WITH_PYRIGHT.md
* https://github.com/blakeNaccarato/boilerdata/issues/58
* https://github.com/blakeNaccarato/pylance-stubs-unofficial
* https://github.com/microsoft/pylance-release/issues/3045#issuecomment-1317402722
* https://github.com/microsoft/pylance-release/discussions/3638

I was able to get pylance and pyright to agree locally, but now in my github action running `pyright` I am seeing errors that I don't see when I run `pyright` locally.

Here's all the context I could compare between the two of them that were the same:

* `pyright --version` `pyright 1.1.326`
* `pyright --verbose` includes 
  * `Loading configuration file at <X>` where X is pointing to the same file
  * `Assuming Python version 3.10`
  * `Found 73 source files`

However, the verbose output was somewhat different, like `Could not resolve source for Y` local but not in GHA or `Could not import Z in file`...

I checked `pyright --help` for more commands that could be helpful for debugging, but I don't see any. 

## Help Request

I need to understand how these two run contexts are different in order to get them in sync. Is there a way to definitively show all the context pyright is using to do it's type checking? 

How else could I debug the difference between these run contexts?

**Comment by erictraut** [comment]
It sounds like you're using the same config file for both local and GHA. The next thing I'd check is the Python environment in both cases. Are you using the same Python version? Do you have the same packages installed in each?

You mentioned that the verbose output `Could not resolve source for Y` appeared locally but not in GHA. Is `Y` a module in your project, or does it refer to a package installed in your Python environment?

What diagnostics differ between the two? Do the differences appear to all stem from import resolution issues?
